tests/test_io_elementwise.py

```diff
@@ -650,3 +650,76 @@ def test_read_sparse_array(
     ad.settings.use_sparse_array_on_read = True
     mtx = ad.io.read_elem(f["mtx"])
     assert issubclass(type(mtx), CSArray)
+
+
+@pytest.mark.parametrize(
+    ("chunks", "expected_chunks"),
+    [((1,), (1,)), ((-1,), (120,)), (None, (25,))],
+    ids=["small", "minus_one_uses_full", "none_uses_ondisk_chunking"],
+)
+@pytest.mark.parametrize("arr", [np.arange(120), np.array(["a"] * 120)])
+def test_chunking_1d_array(
+    store: H5Group | ZarrGroup,
+    arr: np.ndarray,
+    chunks: tuple[int] | None,
+    expected_chunks: tuple[int],
+):
+    write_elem(store, "foo", arr, dataset_kwargs={"chunks": 25})
+    arr = read_elem_lazy(store["foo"], chunks=chunks)
+    assert arr.chunksize == expected_chunks
+
+
+@pytest.mark.parametrize(
+    ("chunks", "expected_chunks"),
+    [
+        ((1, 50), (1, 50)),
+        ((10, -1), (10, 50)),
+        ((10, None), (10, 50)),
+        (
+            None,
+            (
+                25,
+                25,
+            ),
+        ),
+    ],
+    ids=[
+        "small",
+        "minus_one_uses_full",
+        "none_on_axis_uses_full",
```

Comment:
why doesn’t it use ondisk chunking on that axis? is that just how zarr behaves and there’s no reason to change it?

Reply:
yeah, if -1 canonically means “full”, None should be “default” i.e. derive from on-disk, right?

IDK how breaking of a change that would be.

Reply:
Ah ok, no, `None` actually does mean `full axis` in `dask`, so the current behavior is correct! Good to know I wasn't just randomly coding things.  I will add a comment!

Reply:
Hm, it is a bit confusing that one can pass
- `5` or `None` with `None` meaning “as on disk”, or
- `(…, 5)` or `(…, None)`, with `None` meaning “whole axis”

or did I get something wrong?